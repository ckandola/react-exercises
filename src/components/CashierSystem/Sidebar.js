import React, { useState } from 'react';
import { GiHamburgerMenu } from 'react-icons/gi';
import Modal from '../Modal';
import { useGlobalContext } from '../../context';

const Sidebar = () => {
    const [isOpen, setIsOpen] = useState(false);
    const { openModal } = useGlobalContext();

    const toggleIsOpen = () => {
        setIsOpen(!isOpen);
    }

    const printLastReceipt = () => {
        console.log("Your last receipt was printed");
    }
    
    const printLogOffReport = () => {
        console.log("Your log-off report was printed");
    }

    const showWhatsNew = () => {
        openModal();
    }

    const logoff = () => {
        console.log("You have signed out");
    }

    return (
        <section className="pos-sidebar">
            <button className="pos-sidebar-icon" onClick={toggleIsOpen}>
                <GiHamburgerMenu />
            </button>
            
            <aside className={`pos-sidebar-nav${isOpen ? '-show' : ''}`}>
                <button onClick={printLastReceipt}>Print Last Receipt</button>
                <button onClick={printLogOffReport}>Print Log-Off Report</button>
                <button onClick={showWhatsNew}>What's New</button>
                <button onClick={logoff}>Log Off</button>
            </aside>
            
            <Modal>
                <h2>What's New</h2>
                <ul className="pos-sidebar-whats-new">
                    <li>Sidebar bones have been added</li>
                    <li>LogOff logic will be added if I decide to add a login screen</li>
                </ul>
            </Modal>
        </section>
    );
};

export default Sidebar;
