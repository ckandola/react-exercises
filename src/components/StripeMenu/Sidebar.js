import React from 'react';
import { FaTimes } from 'react-icons/fa'
import sublinks from './data';
import { useGlobalContext } from './index';

const Sidebar = () => {
    const { isSidebarOpen, closeSidebar } = useGlobalContext();
    return (
        <aside className={`smenu-sidebar-wrapper ${isSidebarOpen && 'show'}`}>
            <div className="smenu-sidebar">
                <button
                    className="smenu-close-btn"
                    onClick={closeSidebar}>
                        <FaTimes />
                </button>
                <div className="smenu-sidebar-links">
                    {sublinks.map((item, index) => {
                        const { links, page } = item;
                        return (
                            <article key={index}>
                                <h4>{page}</h4>
                                <div className="smenu-sidebar-sublinks">
                                    {links.map((link, index2) => {
                                        return <a key={index2} href={link.url}>
                                            {link.icon}
                                            {link.label}
                                        </a>
                                    })}
                                </div>
                            </article>
                        );
                    })}
                </div>
            </div>
        </aside>
    );
};

export default Sidebar;
