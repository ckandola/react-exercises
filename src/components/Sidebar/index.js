import React, {useState} from 'react';
import './Sidebar.css';
import {FaBars, FaTimes} from 'react-icons/fa';
import { social, links } from './data';
import logo from './logo.svg';

const Sidebar = () => {
    const [isOpen, setIsOpen] = useState(false);

    const toggleSidebar = () => {
        setIsOpen(!isOpen);
    }

    return (
        <div>
            <main className="sidebar-main">
                <button 
                    className="sidebar-toggle"
                    onClick={toggleSidebar}>
                    <FaBars />
                </button>
            </main>

            <aside className={`sidebar ${isOpen && 'show-sidebar'}`}>
                <div className="sidebar-header">
                    <img src={logo} className="sidebar-logo" alt="logo"/>
                    <button 
                        className="sidebar-close-btn"
                        onClick={toggleSidebar}>
                        <FaTimes />
                    </button>
                </div>
                <ul className="sidebar-links">
                    {links.map(link => {
                        return (
                            <li key={link.id}>
                                <a href={link.url}>{link.icon}{link.text}</a>
                            </li>
                        );
                    })}
                </ul>
                <ul className="sidebar-social-icons">
                    {social.map(link => {
                        return (
                            <li key={link.id}>
                                <a href={link.url}>{link.icon}</a>
                            </li>
                        );
                    })}
                </ul>
            </aside>
        </div>
    );
};

export default Sidebar;
