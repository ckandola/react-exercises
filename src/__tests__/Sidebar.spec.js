import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Sidebar from '../components/Sidebar';

describe('Sidebar tests', () => {
    it('matches the snapshot', () => {
        const wrapper = shallow(<Sidebar />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('opens the menu display when the hamburger button is clicked', () => {
        const wrapper = shallow(<Sidebar />);
        expect(wrapper.find('.show-sidebar')).toHaveLength(0);
        wrapper.find('.sidebar-toggle').simulate('click');
        expect(wrapper.find('.show-sidebar')).toHaveLength(1);
    });

    it('closes the menu display when the red X button is clicked', () => {
        const wrapper = shallow(<Sidebar />);
        wrapper.find('.sidebar-toggle').simulate('click');
        expect(wrapper.find('.show-sidebar')).toHaveLength(1);
        wrapper.find('.sidebar-close-btn').simulate('click');
        expect(wrapper.find('.show-sidebar')).toHaveLength(0);
    });
});
