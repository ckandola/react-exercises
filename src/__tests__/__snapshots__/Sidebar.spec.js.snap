// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar tests matches the snapshot 1`] = `
<div>
  <main
    className="sidebar-main"
  >
    <button
      className="sidebar-toggle"
      onClick={[Function]}
    >
      <FaBars />
    </button>
  </main>
  <aside
    className="sidebar false"
  >
    <div
      className="sidebar-header"
    >
      <img
        alt="logo"
        className="sidebar-logo"
        src="logo.svg"
      />
      <button
        className="sidebar-close-btn"
        onClick={[Function]}
      >
        <FaTimes />
      </button>
    </div>
    <ul
      className="sidebar-links"
    >
      <li
        key="1"
      >
        <a
          href="#"
        >
          <FaHome />
          home
        </a>
      </li>
      <li
        key="2"
      >
        <a
          href="#"
        >
          <FaUserFriends />
          team
        </a>
      </li>
      <li
        key="3"
      >
        <a
          href="#"
        >
          <FaFolderOpen />
          projects
        </a>
      </li>
      <li
        key="4"
      >
        <a
          href="#"
        >
          <FaCalendarAlt />
          calendar
        </a>
      </li>
      <li
        key="5"
      >
        <a
          href="#"
        >
          <FaWpforms />
          documents
        </a>
      </li>
    </ul>
    <ul
      className="sidebar-social-icons"
    >
      <li
        key="1"
      >
        <a
          href="https://www.facebook.com"
        >
          <FaFacebook />
        </a>
      </li>
      <li
        key="2"
      >
        <a
          href="https://www.twitter.com"
        >
          <FaTwitter />
        </a>
      </li>
      <li
        key="3"
      >
        <a
          href="https://www.linkedin.com"
        >
          <FaLinkedin />
        </a>
      </li>
      <li
        key="4"
      >
        <a
          href="https://www.behance.net"
        >
          <FaBehance />
        </a>
      </li>
      <li
        key="5"
      >
        <a
          href="https://www.sketch.com"
        >
          <FaSketch />
        </a>
      </li>
    </ul>
  </aside>
</div>
`;
